interface.go

```diff
@@ -171,11 +171,29 @@ type RequestHTTPBuilder interface {
 
 // RequestParams is a scope of methods for configurate http request
 type RequestParams interface {
-	// RequestRepeat is a count of repeat request, if request was failed.
+	// RequestRepeat is a function for set options in request
+	// if response.Code != Expect.Code, than request will repeat counts with delay.
+	// Default delay is 1 second.
 	RequestRepeat(count int) RequestHTTPBuilder
-	// RequestRepeatDelay is a time between repeat request, if request was failed.
-	// Default 1 second
+
+	// RequestRepeatDelay set delay for request repeat.
+	// if response.Code != Expect.Code, than request will repeat counts with delay.
+	// Default delay is 1 second.
 	RequestRepeatDelay(delay time.Duration) RequestHTTPBuilder
+
+	// RequestRepeatPolitic is a politic for repeat request.
+	// if response.Code != Expect.Code, than request will repeat counts with delay.
+	// if Optional is true and request is failed, than test step allure will be option, and t.Fail() will not execute.
+	// If Broken is true and request is failed, than test step allure will be broken, and t.Fail() will execute.
+	RequestRepeatPolitic(politic *RequestRepeatPolitic) RequestHTTPBuilder
+
+	// RequestRepeatOptional is a option politic for repeat request.
+	// if Optional is true and request is failed, than test step allure will be option, and t.Fail() will not execute.
+	RequestRepeatOptional(optional bool) RequestHTTPBuilder
+
+	// RequestRepeatBroken is a broken politic for repeat request.
+	// If Broken is true and request is failed, than test step allure will be broken, and t.Fail() will execute.
```

Comment:
I think that Broken will be very similar to just keep default behaviour then
I was thinking that you introduced the Broken flag to have another way of handling the tests.

Basically I thought as follow:
1. Optional: true -> Will mark the retry as skipped and not fail the whole test
2. Broken: true -> Will mark the retry as failed, the whole test as broken, but not fail the whole test
3. Optional: false, Broken false -> Will mark the retries as failed as well as the overall test as failed

Let me know if my understanding is incorrect

Reply:
Broken true -> Will mark the retries as failed as well as the overall test as failed.

Reply:
@siller174 Sorry for late reply, let me give more details:
![image](https://github.com/ozontech/cute/assets/24391573/c259d977-5e7f-4a00-8594-11ad2c49b7d8)
This is currently what is happening when I put Optional: true

Here is what I had in mind with the optional mechanism:
1. If all retries are failing -> fail the test and do not execute the rest of the test (assertions and so on)
2. If after x retry (x < maxRetryCount), the x retry is successful, continue the rest of the test (assertions and so on) 

Aside from this, for now the retry is mostly based on the HttpCode assertion, but I would say it can apply to all kind of assertions instead (body assertion, http code assertion, header assertion, response assertion)
i.e. if any kind of assertion fails during the test process, apply the same retry logic as mentioned in points 1. and 2. above